dotnetv3/EC2/VirtualPrivateCloudExamples/CreateVPCforS3Example/CreateVPCforS3.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier:  Apache-2.0
+
+
+using Amazon.EC2;
+using Amazon.EC2.Model;
+using Amazon.S3;
+using Amazon.S3.Model;
+using Microsoft.Extensions.Configuration;
+using Tag = Amazon.EC2.Model.Tag;
+
+namespace CreateVPCforS3Example;
+
+/// <summary>
+/// Use a Amazon Elastic Compute Cloud (Amazon EC2) client to create a VPC Endpoint
+/// for the Amazon Simple Storage Service (Amazon S3) service and use
+/// that endpoint to list the objects in the S3 bucket.
+/// </summary>
+public class CreateVPCforS3
+{
+    static async Task Main(string[] args)
+    {
+        var ec2Client = new AmazonEC2Client();
+
+        var configuration = new ConfigurationBuilder()
+            .SetBasePath(Directory.GetCurrentDirectory())
+            .AddJsonFile("appsettings.json") // Load app settings from .json file.
+            .AddJsonFile("appsettings.local.json",
+                true) // Optionally, load local settings.
+            .Build();
+
+        var s3EndpointUrl = await CreateVPCforS3Client(configuration, ec2Client);
+
+        if (s3EndpointUrl != null)
+        {
+            var objectsList = await CreateS3ClientWithEndpoint(configuration, s3EndpointUrl);
+            Console.WriteLine("List of objects in bucket:");
+            Console.WriteLine(string.Join(" ", objectsList.Select(o => o.Key)));
+        }
+    }
+
+
+    // snippet-start:[EC2.dotnetv3.CreateVPCforS3]
+
+    /// <summary>
+    /// Create a VPC Endpoint and use it for an S3 client.
```

Comment:
/// Create a VPC endpoint and use it for an Amazon S3 client.

Reply:
Fixed.